### PR TITLE
Code to unwrap range into list and unit test for it

### LINKED ISSUE #18730: Mantid python does not unwrap range any more
### Current behaviour
 Currently if ones assigns range to a variable in python reduction script, it is unwrapped into list. 

### Actual behaviour 
Some internal changes in mantid python, made for compliance with Python 3, changed this behaviour so if one assigns range to a variable, in fact you assigning an iterator. Direct inelastic reduction scripts fail as expect list or string as input.

Probably best way ahead in this situation (bearing in mind  future changes to Manitd) is to modify script to accept iterator.

### PR DESCRIPTION
**Pull request by abuts**
Fixes #18730

Current ISIS direct reduction scripts fail if one assigns range to it. The changes allow it to work. 

**To test:**

The unit test provided is seems good indicator to see how the changes work. I have tested reduction scripts. Test by code review. 

Users should not notice the changes. No release notes are necessary. 

---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####

- [ ] Is the code of an acceptable quality?
- [ ] Does the code conform to the [coding standards](http://www.mantidproject.org/Coding_Standards)? Is it well structured with small focussed classes/methods/functions?
- [ ] Are there unit/system tests in place? Are the unit tests small and test the a class in isolation?
- [ ] If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- [ ] Do changes function as described? Add comments below that describe the tests performed?
- [ ] How do the changes handle unexpected situations, e.g. bad input?
- [ ] Has the relevant documentation been added/updated?
- [ ] Is user-facing documentation written in a user-friendly manner?
- [ ] Has developer documentation been updated if required?

- Does everything look good? Comment with the ship it emoji but don't merge. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

File: scripts/Inelastic/Direct/RunDescriptor.py [hunk 1]
```diff
@@ -456,6 +456,9 @@ def __set__(self,instance,value):
                     self._set_run_list(instance,run_num,file_path,fext)
                 else:
                     self._set_single_run(instance,run_num,file_path,fext)
+        elif isinstance(value, collections.Iterable): # xrange provided
```
Comment (AndreiSavici) [inline scripts/Inelastic/Direct/RunDescriptor.py]: This is executed for list as well, so the next elif is redundant

File: scripts/test/RunDescriptorTest.py [hunk 2]
```diff
@@ -380,6 +380,12 @@ def test_get_run_list(self):
         run_list = PropertyManager.sample_run.get_run_list()
         self.assertEqual(len(run_list),1)
 
+        # check if run list can be set up using range
+        propman.sample_run = range(1,4)
+        run_list = PropertyManager.sample_run.get_run_list()
+        self.assertEqual(len(run_list),3)
```
Comment (AndreiSavici) [inline scripts/test/RunDescriptorTest.py]: You should check that the result is a list, not other object of length 3.

**Comment by AndreiSavici** [comment]
Also, change the branch you want to merge into from `master` to `release-v3.9`

File: scripts/Inelastic/Direct/RunDescriptor.py (hunk 1 above)
Comment (abuts) [inline scripts/Inelastic/Direct/RunDescriptor.py]: good point.

Comment (AndreiSavici) [inline scripts/Inelastic/Direct/RunDescriptor.py]: Just forcing a list to be a list is OK, so you won't need to create a separate case

**Comment by AndreiSavici** [comment]
Changes look good. @abuts: are you going to change the branch or does someone need to cherry pick the changes to `release-v3.9`?

**Comment by abuts** [comment]
I am closing this pr and move it to pr  #18738, which requests merge into release-3.9

**Pull request closed without merging**